api/crossref/permissions.py

```diff
@@ -0,0 +1,38 @@
+# -*- coding: utf-8 -*-
+import hmac
+import hashlib
+
+from rest_framework import permissions
+from rest_framework import exceptions
+
+from framework import sentry
+from website import settings
+
+
+class RequestComesFromMailgun(permissions.BasePermission):
+    """Verify that request comes from Mailgun.
+    Adapted here from conferences/message.py
+    Signature comparisons as recomended from mailgun docs:
+    https://documentation.mailgun.com/en/latest/user_manual.html#webhooks
+    """
+    def has_permission(self, request, view):
+        if request.method != 'POST':
+            raise exceptions.MethodNotAllowed(method=request.method)
+        data = request.data
+        if not data:
+            raise exceptions.ParseError('Correct headers not returned with request')
```

Comment:
Minor: The error message is inaccurate. We expect a request body, not headers. How about `raise exceptions.ParseError('Request body is empty')`